File: Python3/contains_duplicate_II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        if len(nums) <= 1: return False
        
        hashmap = {}
        
        for i in range(len(nums)):
            if nums[i] in hashmap and abs(i-hashmap[nums[i]]) <= k: 
                return True
            hashmap[nums[i]] = i
            
        return False

File: C++/contains_duplicate_II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int, int> hashmap;
        
        for(int i = 0; i < nums.size(); i++) {
            if(hashmap.find(nums[i]) != hashmap.end() && i-hashmap[nums[i]] <= k){ 
                return true;
            }
            hashmap[nums[i]] = i;
        }
            
        return false;
    }
};
